src/org/openlcb/cdi/swing/CdiPanel.java

```diff
@@ -1010,10 +1058,206 @@ public void propertyChange(PropertyChangeEvent event) {
+                   writeEntry(e.key, org.openlcb.Utilities.toHexDotsString(e.getValue
+                           ().getContents()));
+                }
+                
+                protected void writeEntry(String key, String entry) {
+                    result.append(key);
+                    result.append("=");
+                    // result.append(entry); // use the value currently in CD
+                    result.append(entriesByKey.get(key).getCurrentValue()); // use value currently in UI
+                    result.append("\n");
+                }
+            };
+            
+            visitor.visitGroupRep(rep);
+            return new String(result);
+        }  
+            
+        
+        /**
+         * Perform a "paste" operation into a selected group tab
+         */
+        protected void performGroupReplPaste(int index, ConfigRepresentation.GroupRep rep) {
+            // retrieve from clipboard
+            Clipboard c = Toolkit.getDefaultToolkit().getSystemClipboard();
+            Transferable t = c.getContents( null );
+            String newContentString = "";
+            if ( t.isDataFlavorSupported(DataFlavor.stringFlavor) ) {
+                try {
+                    newContentString = (String)t.getTransferData( DataFlavor.stringFlavor );
+                } catch (UnsupportedFlavorException | IOException e) {
+                    // this can never happen as we checked bafore
+                    return;
+                }
+            } // this should always have succeeded, but if it doesn't the match below will fail
+            
+            // store the values that are going to be replaced
+            String previousContentString = groupReplToString(rep); 
+
+            String[] newContentLines = newContentString.split("\n");
+            String[] previousContentLines = previousContentString.split("\n");
+            
+            // compare keys to see if the variables match up
+            // this version just checks the line count, more could be added here
+            if (previousContentLines.length != newContentLines.length) {
+                logger.log(Level.WARNING, "Cannot paste into a mis-matching entry type");
+                return;
+            }
+            
+            // change the repl number in the newContentLines to this index
+            // First, find the prefix we're going to change
+            List<String> list = java.util.Arrays.asList(newContentLines);
+            String prefix = Utilities.longestLeadingSubstring(list);
+            // That prefix should end with "(1)."
+            
+            // replace and rebuild the lines
+            StringBuilder processedContentSB = new StringBuilder();
+            for (int i = 0; i<newContentLines.length; i++) {
+                newContentLines[i] = newContentLines[i].substring(0, prefix.length()-4)+"("+index+")."+newContentLines[i].substring(prefix.length())+"\n";
```

Comment:
Instead of creating the new prefix by hand, we could use rep.key as the desired prefix.
There are cornercases where this manual edit won't do the right thing, for example when there are two nested repeated groups.